app/types/index.ts

```diff
@@ -1,12 +1,33 @@
+import type { Server } from "socket.io";
+import type { Socket } from "socket.io-client";
 import type { Field } from "@/utils/game";
 
 export type ServerToClientEvents = {
   update(state: Field["plots"]): void;
   clientsCount(count: number): void;
   exposedPercent(count: number): void;
+  session: ({ sessionID }: { sessionID: string }) => void;
+  death(): void;
 };
 
-export type ClientToServerEvents = {
+type ClientToServerEvents = {
   expose(index: number): void;
   flag(index: number): void;
 };
+
+type SocketData = {
+  sessionID: string;
+};
+
+type DefaultEventsMap = {
+  [event: string]: (...args: any[]) => void;
+};
+
+export type SocketClient = Socket<ServerToClientEvents, ClientToServerEvents>;
+
+export type SocketServer = Server<
+  ClientToServerEvents,
+  ServerToClientEvents,
+  DefaultEventsMap,
+  SocketData
+>;
```

Comment:
...that's my bad, this should be `InterServerEvents`
...keeping the order as is used in the generic would please my ocd ❤️
...now that I think about it, it should also be `ClientSocket` and `ServerSocket` too 🙃
```suggestion
type InterServerEvents = {};

type SocketData = {
  sessionID: string;
};

export type ClientSocket = Socket<ServerToClientEvents, ClientToServerEvents>;

export type ServerSocket = Server<
  ClientToServerEvents,
  ServerToClientEvents,
  InterServerEvents,
  SocketData
>;
```

or if you want to be more correct, and keep everything but the name
```suggestion
type InterServerEvents = {
  [event: string]: (...args: any[]) => void;
};

type SocketData = {
  sessionID: string;
};

export type ClientSocket = Socket<ServerToClientEvents, ClientToServerEvents>;

export type ServerSocket = Server<
  ClientToServerEvents,
  ServerToClientEvents,
  InterServerEvents,
  SocketData
>;
```

see: [Socket.IO | TypeScript | Types for the server](https://socket.io/docs/v4/typescript/#types-for-the-server)